src/test/ui/issue-44406.stderr

```diff
@@ -2,7 +2,7 @@ error: expected identifier, found keyword `true`
   --> $DIR/issue-44406.rs:18:10
    |
 18 |     foo!(true); //~ ERROR expected type, found keyword
-   |          ^^^^
+   |          ^^^^ expected identifier, found keyword
```

Comment:
This case looks interesting.
We don't actually expect an identifier here, why does the note appears?

Reply:
A previous commit in a separate PR actually removed this incorrect issue (mostly by accident), but it broke too many things to be mergeable.

[The macro](https://github.com/estebank/rust/blob/d17e38f4bf3418770b7314bad0b8ef4490db55bd/src/test/ui/issue-44406.rs) is expecting a token tree as an argument, so I'm sure it's coming from that. Both cases should probably be pointing at `$rest` instead, like macro errors that happen after parsing.

Reply:
>We don't actually expect an identifier here

My eyes deceive me, we expect an identifier (but this is not related to `tt`).
First we expect a type, but then we try to do recovery in `parse_assoc_op_cast` and get one more message (not entirely desirable) about expecting an identifier.
So, it's ok, not related to this PR.